broker/pkg/carrier/grpcserver.go

```diff
@@ -208,13 +211,14 @@ func (s *GRPCServer) TimelineConsume(stream grpc.Broker_TimelineConsumeServer) e
 			grpc.TimelinePushLeaseResponseAddExpirationTSMSUTC(builder, lease.ExpirationTimestampMS)
 			rootPosition := grpc.TimelinePushLeaseResponseEnd(builder)
 
+			metricTopicLatency.With(prometheus.Labels{"topic": consumerTuple.C.Topic}).Observe(float64(dtime.GetLatencyMS(lease.Message.TimestampMS)))
```

Comment:
for a better accuracy on the Critical Path it should be tracked in the Delete flow, as it is the last part of it.